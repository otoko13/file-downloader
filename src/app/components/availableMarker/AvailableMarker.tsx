import classNames from "classnames";
import { FunctionComponent } from "react";
import styles from "./availableMarker.module.scss";

interface AvailableMarkerProps {
  available?: boolean;
  className?: string;
}

const AvailableMarker: FunctionComponent<AvailableMarkerProps> = ({
  className,
  available,
}) => (
  <div
    role="img"
    aria-label="available"
    aria-hidden={!available}
    className={classNames(className, styles["available-marker"], {
      [styles.visible]: available,
    })}
  />
);

export default AvailableMarker;
