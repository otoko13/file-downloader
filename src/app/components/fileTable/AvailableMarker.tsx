import classNames from "classnames";
import { FunctionComponent } from "react";
import styles from "./availableMarker.module.scss";

interface AvailableMarkerProps {
  available: boolean;
  className?: string;
}

const AvailableMarker: FunctionComponent<AvailableMarkerProps> = ({
  className,
  available,
}) => (
  <div
    className={classNames(className, styles["available-marker"], {
      [styles.visible]: available,
    })}
  />
);

export default AvailableMarker;
