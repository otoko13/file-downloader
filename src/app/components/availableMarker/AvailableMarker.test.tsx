import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import AvailableMarker from "./AvailableMarker";

describe("appearance", () => {
  it("should match the snapshot", () => {
    const { container } = render(<AvailableMarker available />);
    expect(container).toMatchSnapshot();
  });
});

describe("visibility", () => {
  it("should be marked visible when available", () => {
    render(<AvailableMarker available />);
    expect(screen.getByLabelText("available")).toHaveClass("visible");
  });

  it("should be marked hidden when not available", () => {
    render(<AvailableMarker />);
    expect(screen.getByLabelText("available")).not.toHaveClass("visible");
  });
});
