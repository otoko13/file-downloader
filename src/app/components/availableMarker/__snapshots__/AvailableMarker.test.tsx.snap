// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appearance should match the snapshot 1`] = `
<div>
  <div
    aria-hidden="false"
    aria-label="available"
    class="available-marker visible"
    role="img"
  />
</div>
`;
